src/pyhf/tensor/pytorch_backend.py

```diff
@@ -131,35 +123,25 @@ def zeros(self, shape):
         return torch.Tensor(torch.zeros(shape))
 
     def power(self, tensor_in_1, tensor_in_2):
-        tensor_in_1 = self.astensor(tensor_in_1)
-        tensor_in_2 = self.astensor(tensor_in_2)
         return torch.pow(tensor_in_1, tensor_in_2)
 
     def sqrt(self, tensor_in):
-        tensor_in = self.astensor(tensor_in)
         return torch.sqrt(tensor_in)
 
     def divide(self, tensor_in_1, tensor_in_2):
-        tensor_in_1 = self.astensor(tensor_in_1)
-        tensor_in_2 = self.astensor(tensor_in_2)
         return torch.div(tensor_in_1, tensor_in_2)
 
     def log(self, tensor_in):
-        tensor_in = self.astensor(tensor_in)
         return torch.log(tensor_in)
 
     def exp(self, tensor_in):
-        tensor_in = self.astensor(tensor_in)
         return torch.exp(tensor_in)
 
     def stack(self, sequence, axis=0):
         return torch.stack(sequence, dim=axis)
 
     def where(self, mask, tensor_in_1, tensor_in_2):
-        mask = self.astensor(mask).type(torch.FloatTensor)
-        tensor_in_1 = self.astensor(tensor_in_1)
-        tensor_in_2 = self.astensor(tensor_in_2)
-        return mask * tensor_in_1 + (1 - mask) * tensor_in_2
+        return torch.where(mask.type(torch.BoolTensor), tensor_in_1, tensor_in_2)
```

Comment:
@kratsg Your comment got removed when I combined the two lines, so answering here again:

> why is there a cast here? The mask should be `bool` already.

I think that it is better to just cast it to make sure. There are times like the following where the mask seems like a valid `Bool`

https://github.com/diana-hep/pyhf/blob/c8a94f4d5a11edfdd5a28184153e1ef4b076e638/tests/test_tensor.py#L48-L50

but TensorFlow doesn't see it as such if the `tf.cast(mask, tf.bool)` is removed. So I think that it should just be used to ensure the right Tensor `dtype`.